lxd/db/node.go

```diff
@@ -138,13 +141,21 @@ func (n NodeInfo) ToAPI(cluster *Cluster, node *Node) (*api.ClusterMember, error
 	result.Database = false
 	result.Config = n.Config
 	result.Roles = n.Roles
-	if raftNode != nil && raftNode.Role == RaftVoter {
-		result.Roles = append(result.Roles, string(ClusterRoleDatabase))
-		result.Database = true
-	}
-	if raftNode != nil && raftNode.Role == RaftStandBy {
-		result.Roles = append(result.Roles, string(ClusterRoleDatabaseStandBy))
+
+	// Check if node is the leader node
+	if leader == n.Address {
```

Comment:
Slight tweak here, I think we want the leader role in addition to the database role as the leader is still one of the 3 database voters and not listing `database` as a role on it may cause confusion for existing clients.